### PR TITLE
Fix liver and stomach tracking in WereProfessor

### PR DESCRIPTION
**Pull request by burningbman**
Correctly track the stomach and liver capacity in the WereProfessor path. I used License to Adventure as a starting template, but the implementation had to be different since in WereProfessor already selected upgrades are not shown as options in the table.

I'll start poking around KoLCharacterTest and ChoiceControlTest to get some tests for the changes, but I wanted to get feedback on the implementation first.

### CODE REVIEW

**Review by jaadams5 — APPROVED** [review]
I'll approve, in the spirit of so far so good.  I'd love to see some tests but if this gets committed without them I'd be OK with that since supporting new content in a timely fashion is important.

**Comment by burningbman** [comment]
After starting to add tracking for the other modifiers, I'm re-thinking having one preference per upgrade. I think it would be better to just have one stomach/liver preference that is set to 0-3 instead of 3 separate preferences set to true.

**Comment by jaadams5** [comment]
I know nothing about the path but it seems like moving to 12 distinct preferences is a step backwards :-)  However if that is necessary the implementation looks reasonable.

Were you planning on adding more parameterized tests in WereProfessorResearch?  

I see you may have fixed the failing test...

**Comment by midgleyc** [comment]
By wiki comments you should also be able to read the non-liver / stomach effects from the intrinsic descriptions.

**Comment by burningbman** [comment]
Reduced the scope back to just tracking stomach and liver in 2 preferences. Sorry for all the unnecessary flailing.

File: src/data/modifiers.txt [hunk 1]
```diff
@@ -12727,6 +12727,7 @@ Path	License to Adventure	Weapon Damage Percent: [50*pref(bondWpn)], Initiative:
 Path	One Crazy Random Summer	Random Monster Modifiers: +1
 Path	Quantum Terrarium	Familiar Weight: [min(2*L,20)]
 Path	You, Robot	Energy: +1
+Path	WereProfessor	Liver Capacity: [3*(pref(wereLiver))], Stomach Capacity: [3*(pref(wereStomach))]
```
Comment (midgleyc) [inline src/data/modifiers.txt:12730]:
```suggestion
Path	WereProfessor	Liver Capacity: [3*(pref(wereProfessorLiver))], Stomach Capacity: [3*(pref(wereProfessorStomach))]
```

File: src/net/sourceforge/kolmafia/session/ChoiceControl.java [hunk 2]
```diff
@@ -8599,6 +8599,36 @@ else if (text.contains("Order a blueberry muffin")) {
...
+            && !text.contains("Thromboxane inhibitor")) {
+          wereStomach += !text.contains("Triiodothyronine accelerator") ? 1 : 0;
+          wereStomach += wereStomach == 1 && !text.contains("Thyroxine supplements") ? 1 : 0;
+          wereStomach += wereStomach == 2 && !text.contains("Amyloid polypeptide mixture") ? 1 : 0;
+        }
+        Preferences.setInteger("wereProfessorStomach", wereStomach);
+
+        // calculate liver
+        int wereLiver = 0;
+        if (text.contains("Dopamineslurry")
```
Comment (midgleyc) [inline src/net/sourceforge/kolmafia/session/ChoiceControl.java:8620]:
```suggestion
        if (!text.contains("Dopamine slurry")
```

File: src/net/sourceforge/kolmafia/session/ChoiceControl.java [hunk 3]
```diff
@@ -8599,6 +8599,36 @@ else if (text.contains("Order a blueberry muffin")) {
...
+            && !text.contains("Relaxin Balm")
+            && !text.contains("Melatonin suppositories")
+            && !text.contains("Adrenal decoction")
+            && !text.contains("Adrenal distillate")
+            && !text.contains("Concentrated adrenaline extract")) {
+          wereLiver += !text.contains("Glicagon condensate") ? 1 : 0;
+          wereLiver += wereLiver == 1 && !text.contains("Secretin agonist") ? 1 : 0;
+          wereLiver += wereLiver == 1 && !text.contains("Synthetic aldosterone") ? 1 : 0;
```
Comment (midgleyc) [inline src/net/sourceforge/kolmafia/session/ChoiceControl.java:8628]:
```suggestion
          wereLiver += wereLiver == 2 && !text.contains("Synthetic aldosterone") ? 1 : 0;
```

File: test/net/sourceforge/kolmafia/session/ChoiceControlTest.java [hunk 4]
```diff
@@ -504,4 +504,37 @@ void handlesExtracting() {
       }
     }
   }
+
+  @Nested
+  class WereProfessorResearch {
+    @ParameterizedTest
+    @CsvSource({"2, test_choice_wereprofessor_no_liver_two_stomach.html"})
```
Comment (midgleyc) [inline test/net/sourceforge/kolmafia/session/ChoiceControlTest.java:511]: I don't think there's much point having a parameterized test with exactly one set of parameters.

**Review by jaadams5 — APPROVED** [review]
Looks OK.  Push this now or wait for the additional tests?

File: test/root/request/test_choice_wereprofessor_no_liver_two_stomach.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+
+<!-- saved from url=(0047)http://127.0.0.1:60080/choice.php?forceoption=0 -->
```
Comment (midgleyc) [inline test/root/request/test_choice_wereprofessor_no_liver_two_stomach.html:2]: either use this file or remove it?

**Review by midgleyc — APPROVED** [review]

**Pull request merged**